modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCPrimaryKeyFinderViewTestSetup.java

```diff
@@ -0,0 +1,70 @@
+package org.geotools.jdbc;
+
+import java.sql.SQLException;
+
+public abstract class JDBCPrimaryKeyFinderViewTestSetup extends JDBCDelegatingTestSetup {
+
+    protected JDBCPrimaryKeyFinderViewTestSetup(JDBCTestSetup delegate) {
+        super(delegate);
+    }
+
+    @Override
+    protected final void setUpData() throws Exception {
+        // kill all the data
+        try {
+            dropMetadataView();
+        } catch (SQLException e) {
+        }
+        try {
+            dropMetadataTable();
+        } catch (SQLException e) {
+        }
+        try {
+            dropSequencedPrimaryKeyTable();
+        } catch (SQLException e) {
+        }
+
+        // create all the data
+        createMetadataTable();
+        createSequencedPrimaryKeyTable();
+        createMetadataView();
+    }
+
+    /** Drops the metadata table */
+    protected abstract void dropMetadataTable() throws Exception;
+
+    /** Drops the metadata view */
+    protected abstract void dropMetadataView() throws Exception;
+
+    /** Drops the "seqtable" table. */
+    protected abstract void dropSequencedPrimaryKeyTable() throws Exception;
+
+    /**
+     * Creates the default geotools metadata table, empty.
+     *
+     * <p>The table is named "gt_pk_metdata". See {@link MetadataTablePrimaryKeyFinder} javadoc for
+     * the expected table structure
+     */
+    protected abstract void createMetadataTable() throws Exception;
+
+    /**
+     * Creates a table named 'plaintable' with no primary key and the following structure
```

Comment:
The javadoc of a setup method has to explain the actual work that needs doing. In this case, creating a view named ``gt_pk_metadata`` based on the table created by ``createMetadataTable``

Reply:
Done.